presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -862,24 +868,46 @@ public static Slice stEndPoint(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     }
 
     @SqlNullable
-    @Description("Returns an array of points in a linestring")
+    @Description("Returns an array of points in a geometry")
     @ScalarFunction("ST_Points")
     @SqlType("array(" + GEOMETRY_TYPE_NAME + ")")
     public static Block stPoints(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     {
         Geometry geometry = deserialize(input);
-        validateType("ST_Points", geometry, EnumSet.of(LINE_STRING));
+        validateType("ST_Points", geometry, VALID_TYPES_FOR_ST_POINTS);
         if (geometry.isEmpty()) {
             return null;
         }
-        LineString lineString = (LineString) geometry;
-        BlockBuilder blockBuilder = GEOMETRY.createBlockBuilder(null, lineString.getNumPoints());
-        for (int i = 0; i < lineString.getNumPoints(); i++) {
-            GEOMETRY.writeSlice(blockBuilder, serialize(lineString.getPointN(i)));
-        }
+
+        int pointCount = geometry.getNumPoints();
```

Comment:
You can inline the pointCount variable, since it's only used in one place:
```java
BlockBuilder blockBuilder = GEOMETRY.createBlockBuilder(null, geometry.getNumPoints());
```

Reply:
Thanks @jagill. I removed pointCount.